rest_framework/permissions.py

```diff
@@ -226,8 +226,14 @@ def has_permission(self, request, view):
            not request.user.is_authenticated and self.authenticated_users_only):
             return False
 
-        queryset = self._queryset(view)
-        perms = self.get_required_permissions(request.method, queryset.model)
+        if hasattr(view, 'perms_map') and request.method in view.perms_map:
+            # Override default perms_map from view / ViewSet.
+            perms = view.perms_map[request.method]
+            if isinstance(perms, str):
+                perms = [perms]
```

Comment:
I don't see what changes here as opposed to calling `get_required_permissions`

Reply:
In case supplied view / viewset has `perms_map` attribute defined, it would not call `get_required_permissions` at all. This has the following benefits:

1. `DjangoModelPermissions` is not coupled / tied to the queryset. Queryset becomes optional, not mandatory one.
2. There is no need to define new `Permissions` class just to have custom `perms_map` values.

Reply:
Right, I missed the point that `perms_map` is extracted from the view.
I'll be -1 on that part for several reasons.
1. It would define the permissions map at two different locations which is something I'd like to avoid
2. How would that work with actions on that view ?
3. If - and I say IF - we'd go down that route, what would the plan be for the migrations - to avoid duplicated permissions map
4. Overriding the Permission to redefine the permissions map is more explicit to me than a complex permission check, let's keep things simple if possible.

Reply:
I think a permissions class that pulls the `perms_map` from the view is a good idea. However, I think it should be an entirely new permissions class. This would avoid some of the issues @xordoquy raised.

Reply:
I am a total newbie with DRF. I just had freelance project, where I was asked to develop custom Django auth backend. Auth backend works but the developers complained that it does not fit DRF well.

I just checked DRF actions, it seems to use "if" to instantiate Permission class depending on action name.

The idea that separate actions should have different Permission class and the idea that just to change `perms_map` sting value one should extend the class, is a very strange one. It would create many small classes which do nothing besides re-defined `perms_map` attribute value.

Note that Django does not require to define many auth backends for different views, it works with one auth backend. That is much cleaner architecture.

In case actions use `perms_map` I guess it should have nested format like this:

```
perms_map = {
        'retrieve': {
            'GET': ['%(app_label)s.view_%(model_name)s'],
        }
}
```

Reply:
Reported that standard Django `@permission_required` decorator should work with DRF action methods via `@method_decorator`:

`@method_decorator(permission_required('my_app.view_model',raise_exception=True))`